### PR TITLE
Add plugin to allow triggering rsync configurations from Tasker / Locale

### PR DESCRIPTION
**Pull request by graemechapman**
Thought this would be useful to add a bit more flexibility in triggering tasks as an alternative to using schedules. 

I am also a Java / Android novice so any feedback / criticism is welcome!

### CODE REVIEW

**Comment by socratesx** [comment]
Thank you for your contribution. This plugin will enhance the app's usability. Please allow me some time to review your changes.

File: app/src/main/java/com/linminitools/myrsync/PluginBundleValues.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    /**
+     * @param context Application context.
+     * @param configId The id of the config file selected.
+     * @return A plug-in bundle.
+     */
+    @NonNull
+    public static Bundle generateBundle(@NonNull final Context context,
+                                        @NonNull final Integer configId) {
+        assertNotNull(context, "context"); //$NON-NLS-1$
+//        assertNotEmpty(configId, "configId"); //$NON-NLS-1$
```
Comment (mades-g) [inline app/src/main/java/com/linminitools/myrsync/PluginBundleValues.java]: Maybe we can remove this?

Comment (graemechapman) [reply app/src/main/java/com/linminitools/myrsync/PluginBundleValues.java]: Ooh, don't think that was meant to be commented out!

**Comment by socratesx** [comment]
Hello,

I saw your changes and I will accept them. Please, since I don't have the tasker application installed, can you describe how the user can use myrsync along with the tasker? What are the logic steps, the user must perform in order to run a configuration according to a tasker's condition? An example would help.
Thank you!

Regards,
Socrates

**Comment by graemechapman** [comment]
Hi @socratesx, many thanks for your quick review and for accepting my changes. 

Absolutely, I will write out a description for you soon, I can provide some screenshots too if you like? I believe there is a free trial version of Tasker available if you would like to try it out for yourself.

**Comment by graemechapman** [comment]
Hi @socratesx,

From a Tasker task, under the `plugins` heading, there is a `myRSync` action. The configuration option will take you into myRSync which will allow you to select a configuration, from the list of configs already defined within the app. Simply select a configuration and then press back to save. You can then trigger the rsync task directly from a profile defined within Tasker, e.g. on connection to a specific WiFi SSID.

I have attached a few screenshots below to help illustrate.
screenshot_1547902681
screenshot_1547902703
screenshot_1547902865
screenshot_1547902877

**Comment by socratesx** [comment]
Hi @graemechapman

I tested it out and it is working perfectly. Thank you very much for this useful addition!

Best Regards,
Socrates

**Comment by graemechapman** [comment]
You're welcome :slightly_smiling_face:

**Pull request merged**